provider/kv.go

```diff
@@ -60,17 +86,19 @@ func (provider *Kv) watchKv(configurationChan chan<- types.ConfigMessage, prefix
 					}
 				}
 			}
+			return nil
 		}
-	}
 
-	notify := func(err error, time time.Duration) {
-		log.Errorf("KV connection error: %+v, retrying in %s", err, time)
-	}
-	err := backoff.RetryNotify(operation, backoff.NewExponentialBackOff(), notify)
-	if err != nil {
-		return fmt.Errorf("Cannot connect to KV server: %v", err)
+		notify := func(err error, time time.Duration) {
+			log.Errorf("KV watch connection error: %+v, retrying in %s", err, time)
+		}
+		b := backoff.NewExponentialBackOff()
+		b.MaxElapsedTime = 0 // Keep backing off forever
```

Comment:
This change is a bit controversial. My feelings: In the event that an etcd cluster goes down (or there's a network partition, etc.) it's going to take us longer than `DefaultMaxElapsedTime` (10 minutes) to get the cluster back up. Given that we're logging with the `notify` func here, is it harmful to have traefik backoff/retry indefinitely?